Library/Homebrew/livecheck/strategy.rb

```diff
@@ -75,6 +75,9 @@ module Strategy
       # In rare cases, this can also be a double newline (`\n\n`).
       HTTP_HEAD_BODY_SEPARATOR = "\r\n\r\n"
 
+      # A regex used to identify a tarball extension at the end of a string.
+      TARBALL_EXTENSION_REGEX = /\.t(?:ar\..+|[a-z0-9]+)$/i.freeze
```

Comment:
I was quietly hoping to improve this later but, now that you mentioned it, I'll take care of it here, ahah. After rebasing this and resolving the conflicts, I added a commit that uses a more explicit regex which will only match specific tarball extensions.

The upside is that this will only match what it should (e.g., not `.txt`) but the downside is that it will need to be kept up to date if/when new extensions appear. In that scenario, a given strategy regex will simply use the full tarball extension instead of `.t` until the `TARBALL_EXTENSION_REGEX` is updated. In most cases, this shouldn't be a problem, as it would likely omit older versions instead of newer ones (e.g., if a given source starts publishing tarballs in a new format). I also don't foresee this being a common occurrence, so I'm not too worried.

The alternative to this is to expand the `UnpackStrategy::Tar#extensions` array and use that to programmatically generate this regex constant. That approach would avoid duplicating this information but I wasn't sure whether updating the `#extensions` array for this purpose would be appropriate. The main question is whether the `#extensions` array is supposed to represent supported extensions or any possible tarball extension. `#can_extract` would return `false` for a tarball with an unsupported extension (so it may be fine) but the intention of the `#extensions` array was ambiguous to me, so I simply erred on the side of using a regex literal here.

Reply:
Thanks for fixing the regex.

> The main question is whether the `#extensions` array is supposed to represent supported extensions or any possible tarball extension.

Given that there is `can_extract?` method, I assume that the `extensions` method lists all possible tarball extensions.

Reply:
That seems reasonable to me. I'll revisit this idea when I get a chance and we can definitively determine whether updating `Tar#extensions` is appropriate in a PR. It shouldn't be much work, so it would be fine if the answer ends up being "no".